android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,90 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.theme.Typography
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState {
+    class InputAsTextState(val text: String) : MovieCategoryLabelTextViewState()
+    class InputAsResourceState(@StringRes val textRes: Int) : MovieCategoryLabelTextViewState()
+}
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int, val isSelected: Boolean, val categoryText: MovieCategoryLabelTextViewState
+)
+
+@Composable
+fun MovieCategoryLabel(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState,
+    modifier: Modifier = Modifier,
+    onTextClick: () -> Unit = {},
+) {
+    Column(modifier = modifier
+        .width(IntrinsicSize.Max)
+        .clickable { onTextClick() }) {
+        if (movieCategoryLabelViewState.isSelected) {
+            Text(
+                text = selectInputType(movieCategoryLabelViewState = movieCategoryLabelViewState),
+                style = Typography.h3,
+                maxLines = 2,
+                overflow = TextOverflow.Ellipsis
+            )
+
+            Divider(
+                color = Color.Black, thickness = 2.dp, modifier = Modifier.fillMaxWidth()
+            )
+        } else {
+            UnselectedText(movieCategoryLabelViewState = movieCategoryLabelViewState)
+        }
+    }
+}
+
+@Composable
+fun UnselectedText(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState, modifier: Modifier = Modifier
+) {
+    Text(
+        text = selectInputType(movieCategoryLabelViewState = movieCategoryLabelViewState),
+        style = Typography.h4,
+        fontSize = 16.sp,
+        maxLines = 2,
+        overflow = TextOverflow.Ellipsis,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun selectInputType(
```

Comment:
All Compose functions must start with capital letter.

Reply:
Android studio gives me this warning: Composable functions with a return type should start with a lowercase letter. Was it not suppose to have a return type?

Reply:
I just checked the code standards and you are right composable functions that return non-Unit type should follow standard kotlin conventions for function naming. Sorry for the incorrect instructions and thank you for correcting me.